File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/debugging/DebugExample.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 
Imports System
Imports System.IO
Imports com.db4o
Imports com.db4o.query

Namespace com.db4odoc.f1.debugging
    Public Class DebugExample
        Public Shared ReadOnly YapFileName As String = "formula1.yap"

        Public Shared Sub Main(ByVal args() As String)
            SetCars()
        End Sub
        ' end Main

        Public Shared Sub SetCars()
            Db4o.Configure().MessageLevel(3)
            File.Delete(YapFileName)
            Dim db As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                Dim car1 As Car = New Car("BMW")
                db.Set(car1)
                Dim car2 As Car = New Car("Ferrari")
                db.Set(car2)
                db.Deactivate(car1, 2)
                Dim query As Query = db.Query()
                query.Constrain(GetType(Car))
                Dim results As ObjectSet = query.Execute()
                ListResult(results)
            Finally
                db.Close()
            End Try
            Db4o.Configure().MessageLevel(0)
        End Sub
        ' end SetCars

        Public Shared Sub ListResult(ByVal result As ObjectSet)
            Console.WriteLine(result.Count)
            For Each item As Object In result
                Console.WriteLine(item)
            Next
        End Sub
        ' end ListResult
    End Class
End Namespace

File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/debugging/DebugExample.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */

using System;
using System.IO;
using com.db4o;
using com.db4o.query;
using com.db4odoc.f1;
using com.db4odoc.f1.evaluations;

namespace com.db4odoc.f1.debugging
{
	public class DebugExample
	{
		public readonly static string YapFileName = "formula1.yap";

		public static void Main(String[] args) 
		{
			SetCars();
		}
		// end Main

		public static void SetCars()
		{
			Db4o.Configure().MessageLevel(3);
			File.Delete(YapFileName);
			ObjectContainer db=Db4o.OpenFile(YapFileName);
			try 
			{
				Car car1 = new Car("BMW");
				db.Set(car1);
				Car car2 = new Car("Ferrari");
				db.Set(car2);
				db.Deactivate(car1,2);
				Query query = db.Query();
				query.Constrain(typeof(Car));
				ObjectSet results = query.Execute();
				ListResult(results);
			} 
			finally 
			{
				db.Close();
			}
			Db4o.Configure().MessageLevel(0);
		}
		// end SetCars

		public static void ListResult(ObjectSet result)
		{
			Console.WriteLine(result.Count);
			foreach (object item in result)
			{
				Console.WriteLine(item);
			}
		}
		// end ListResult
	}
}
